staging/src/k8s.io/client-go/rest/config.go

```diff
@@ -275,11 +274,6 @@ func DefaultKubernetesUserAgent() string {
 // running inside a pod running on kubernetes. It will return an error if
 // called from a process not running in a kubernetes environment.
 func InClusterConfig() (*Config, error) {
-	host, port := os.Getenv("KUBERNETES_SERVICE_HOST"), os.Getenv("KUBERNETES_SERVICE_PORT")
```

Comment:
I was hoping we could switch `KUBERNETES_SERVICE_HOST` to be set to `kubernetes.default.svc.cluster.local`, but maybe too many things assume it is an IP today